appsDiscovery/src/main/java/org/kinecosystem/appsdiscovery/receiver/presenter/AccountInfoPresenter.java

```diff
@@ -118,61 +124,67 @@ private void startAccountInfoTask(IAccountInfo accountInfo) {
         }
     }
 
-    private void onError() {
+    private void onError(String errorMessage) {
         if (accountInfoResponder != null) {
-            accountInfoResponder.respondError();
+            accountInfoResponder.respondError(errorMessage);
             if (getView() != null) {
                 getView().close();
             }
         }
     }
 
-    private void onTaskComplete(Integer state) {
-        taskState = state;
+    private void onTaskComplete(TaskResponse response) {
+        taskResponse = response;
         if (!isPaused) {
             checkTaskState();
         }
     }
 
     private void checkTaskState() {
         IAccountInfoView accountInfoView = getView();
+        int taskState = taskResponse.taskState;
         if (taskState == TASK_STATE_SUCCESS) {
             if (accountInfoView != null) {
                 accountInfoView.enabledAgreeButton();
             }
-            taskState = TASK_STATE_UNDEFINED;
+            taskResponse.taskState = TASK_STATE_UNDEFINED;
```

Comment:
@berryve is task state here is undefined?  or success?

Reply:
I think the idea here was to reset the state back to undefined (after success has been handled) so that it is not handled more than once.

Reply:
yes yes i remember it was reset ... thanks